clippy_lints/src/implicit_saturating_sub.rs

```diff
@@ -0,0 +1,152 @@
+
+            // Ensure that the binary operator is >, != and <
+            if BinOpKind::Ne == cond_op.node || BinOpKind::Gt == cond_op.node || BinOpKind::Lt == cond_op.node;
+
+            // Check if the true condition block has only one statement
+            if let ExprKind::Block(ref block, _) = then.kind;
+            if block.stmts.len() == 1;
+
+            // Check if assign operation is done
+            if let StmtKind::Semi(ref e) = block.stmts[0].kind;
+            if let ExprKind::AssignOp(ref op1, ref target, ref value) = e.kind;
+            if BinOpKind::Sub == op1.node;
+            if let ExprKind::Path(ref assign_path) = target.kind;
+
+            // Extracting out the variable name
+            if let QPath::Resolved(_, ref ares_path) = assign_path;
+
+            // Get the literal being subtracted
+            if let ExprKind::Lit(ref lit1) = value.kind;
+            if let LitKind::Int(1, _) = lit1.node;
+
+            then {
+                // Handle symmetric conditions in the if statement
+                let (cond_var, cond_num_val) = if SpanlessEq::new(cx).eq_expr(cond_left, target) {
+                    if BinOpKind::Gt == cond_op.node || BinOpKind::Ne == cond_op.node {
+                        (cond_left, cond_right)
+                    } else {
+                        return;
+                    }
+                } else if SpanlessEq::new(cx).eq_expr(cond_right, target) {
+                    if BinOpKind::Lt == cond_op.node || BinOpKind::Ne == cond_op.node {
+                        (cond_right, cond_left)
+                    } else {
+                        return;
+                    }
+                } else {
+                    return;
+                };
+
+                // Check if the variable in the condition statement is an integer
+                if !cx.tables.expr_ty(cond_var).is_integral() {
+                    return;
+                }
+
+                // Get the variable name
+                let var_name = ares_path.segments[0].ident.name.as_str();
+
+                match cond_num_val.kind {
+                    ExprKind::Lit(ref cond_lit) => {
+                        // Check if the constant is zero
+                        if let LitKind::Int(0, _) = cond_lit.node {
+                            if cx.tables.expr_ty(cond_left).is_signed() {
+                                return;
+                            } else {
+                                print_lint_and_sugg(cx, &var_name, expr);
+                            }
+                        } else {
+                            return;
+                        }
+                    },
+                    ExprKind::Path(ref cond_num_path) => {
+                        if match_qpath(cond_num_path, &["i8", "MIN"]) || match_qpath(cond_num_path, &["i16", "MIN"]) || match_qpath(cond_num_path, &["i32", "MIN"]) || match_qpath(cond_num_path, &["i64", "MIN"]) {
```

Comment:
```suggestion
                        if ["i8", "i16", "i32", "i64", "i128"].iter().any(|int_type| match_qpath(cond_num_path, &[int_type, "MIN"])) {
```

or better: extract the array `const INT_TYPES: [&str; 5] = ["i8", "i16", "i32", "i64", "i128"]` and reuse it below.